app/models/entities/item.rb

```diff
@@ -0,0 +1,46 @@
+require 'bigdecimal'
+
+class Entities::Item < Maestrano::Connector::Rails::Entity
+  def self.connec_entity_name
+    'Item'
+  end
+
+  def self.external_entity_name
+    'Product'
+  end
+
+  def self.mapper_class
+    ItemMapper
+  end
+
+  def self.object_name_from_connec_entity_hash(entity)
+    "#{entity['name']}"
+  end
+
+  def self.object_name_from_external_entity_hash(entity)
+    "#{entity['name']}"
+  end
+end
+
+class ItemMapper
+  extend HashMapper
+
+  after_normalize do |input, output|
+    output['sku'] = input['reference'].present? ? input['reference'] : "NOSKU-#{(1..9).to_a.shuffle[0,5].join}"
```

Comment:
you can use input['code'] is there is no reference as there will always be a code